tests/Metalama.Framework.Tests.UnitTests/DesignTime/Pipeline/DesignTimePipelineTests.cs

```diff
@@ -1566,4 +1567,65 @@ class Fabric : TypeFabric
 
         Assert.True( factory.TryExecute( testContext.ProjectOptions, compilation, default, out _ ) );
     }
+
+    [Fact]
+    public async Task HasAspectInEligibilityFailsAsync()
+    {
+        using var testContext = this.CreateTestContext();
+
+        const string code =
+            """
+            using System;
+            using System.Linq;
+            using Metalama.Framework.Aspects;
+            using Metalama.Framework.Code;
+            using Metalama.Framework.Eligibility;
+
+            class Aspect : OverrideMethodAspect
+            {
+                public override void BuildEligibility(IEligibilityBuilder<IMethod> builder)
+                {
+                    builder.MustSatisfy(method =>
+                    {
+                        var hasAspectEnhancements = method.DeclaringType.Methods.OfName("HasAspect").Single().Enhancements();
+
+                        if (!hasAspectEnhancements.HasAspect<Aspect>())
+                        {
+                            throw new Exception();
+                        }
+
+                        return true;
+                    }, _ => $"");
+                }
+
+                public override dynamic? OverrideMethod()
+                {
+                    throw new NotImplementedException();
+                }
+            }
+
+            class TargetCode
+            {
+                private int Method(int a)
+                {
+                    return a;
+                }
+
+                [Aspect]
+                private void HasAspect() { }
+            }
+            """;
+
+        var compilation = CreateCSharpCompilation( new Dictionary<string, string>() { { "code.cs", code } } );
+
+        var methodSymbol = compilation.GetSymbolsWithName( "Method" ).OfType<IMethodSymbol>().Single();
+
+        using TestDesignTimeAspectPipelineFactory factory = new( testContext );
+
+        var pipeline = factory.GetOrCreatePipeline( testContext.ProjectOptions, compilation );
+
+        var exception = await Assert.ThrowsAsync<UserCodeException>( async () => await pipeline.ExecuteAsync( compilation, AsyncExecutionContext.Get() ) );
```

Comment:
To test `PipelineResultBasedAspectRepository`, I would first need to call `pipeline.Execute(...)` to prepare and then `pipeline.GetEligibleAspects(...)` to actually test it.

But `HasAspect` already doesn't work correctly in `Execute`, so there is no way to test `GetEligibleAspects`.